libglusterfs/src/glusterfs/mem-pool.h

```diff
@@ -64,18 +71,19 @@ struct mem_header {
     uint32_t type;
     size_t size;
     struct mem_acct *mem_acct;
-    uint32_t magic;
+    gf_mem_magic_t magic;
 #ifdef DEBUG
     struct list_head acct_list;
 #endif
-    int padding[8];
+    /* ensures alignment */
+    void *data[];
 };
 
 #define GF_MEM_HEADER_SIZE (sizeof(struct mem_header))
 
 #ifdef DEBUG
 struct mem_invalid {
-    uint32_t magic;
+    gf_mem_magic_t magic;
```

Comment:
Why not re-order variables, so things will be nicely aligned? For example, move the time after magic?

Reply:
Does it matter for this one? DEBUG? But I could...

For the main mem_header, re-order to something like this?:

struct mem_header {
-    uint32_t type;
-    size_t size;
     struct mem_acct *mem_acct;
+    size_t size;
+    uint32_t type;
     gf_mem_magic_t magic;
 #ifdef DEBUG
     struct list_head acct_list;
     
?